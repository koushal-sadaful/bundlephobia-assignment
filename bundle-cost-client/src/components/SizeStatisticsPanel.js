import {Card, Grid, Statistic} from "semantic-ui-react";
import React from "react";
import {prettifySize} from "../utils/sizeHelper";


const SizeStatisticsPanel = (props) => {

    return (
        <div className="statistics-panel">
            <Card.Group>
                <Card>
                    <Card.Content>
                        <Statistic>
                            <Statistic.Value>{prettifySize(props.minifiedSize)}</Statistic.Value>
                        </Statistic>
                        <Card.Header>MINIFIED</Card.Header>
                    </Card.Content>
                </Card>
                <Card>
                    <Card.Content>
                        <Statistic>
                            <Statistic.Value>{prettifySize(props.minifiedAndGzipedSize)}</Statistic.Value>
                        </Statistic>
                        <Card.Header>MINIFIED + GZIPPED</Card.Header>
                    </Card.Content>
                </Card>
                <Card>
                    <Card.Content>
                        <Statistic>
                            <Statistic.Value>{prettifySize(props.unpackedSize)}</Statistic.Value>
                        </Statistic>
                        <Card.Header>UNPACKED</Card.Header>
                    </Card.Content>
                </Card>
            </Card.Group>
        </div>
    );
};

export default SizeStatisticsPanel