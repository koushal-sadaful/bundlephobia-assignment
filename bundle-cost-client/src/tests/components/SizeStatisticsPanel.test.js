import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {configure, mount} from 'enzyme';
import {Card, Statistic} from "semantic-ui-react";
import SizeStatisticsPanel from "../../components/SizeStatisticsPanel";
import {runTheSnapshotTest} from "../testUtils/TestHelpers";
configure({adapter: new Adapter()});

describe("SizeStatisticsPanel", () => {
    let wrapper = null;
    const defaultTestProps = {
        minifiedSize: 2115154,
        minifiedAndGzipedSize: 1452635,
        unpackedSize: 41545421
    };

    afterEach(() => {
        if (wrapper){
            wrapper.unmount()
            wrapper = null
        }
    });

    it("should render all the sizes in pretty format", () => {
        wrapper = mount(<SizeStatisticsPanel {...defaultTestProps} />);

        const allHeaderValues = wrapper.find(Card.Header).map(it => it.props().children);
        const allStatsValues = wrapper.find(Statistic.Value).map(it => it.props().children);

        expect(allHeaderValues).toEqual(['MINIFIED', 'MINIFIED + GZIPPED', 'UNPACKED']);
        expect(allStatsValues).toEqual(["2.12 MB", "1.45 MB", "41.5 MB",]);
    });

    it("should handle missing sizes", () => {
        wrapper = mount(<SizeStatisticsPanel {...{}} />);

        const allHeaderValues = wrapper.find(Card.Header).map(it => it.props().children);
        const allStatsValues = wrapper.find(Statistic.Value).map(it => it.props().children);

        expect(allHeaderValues).toEqual(['MINIFIED', 'MINIFIED + GZIPPED', 'UNPACKED']);
        expect(allStatsValues).toEqual(["-", "-", "-",]);
    });

    it("should handle wrong type of sizes", () => {
        const testProps = {
            minifiedSize: "abx",
            minifiedAndGzipedSize: 1452635,
            unpackedSize: "41545421"
        };
        wrapper = mount(<SizeStatisticsPanel {...testProps} />);

        const allHeaderValues = wrapper.find(Card.Header).map(it => it.props().children);
        const allStatsValues = wrapper.find(Statistic.Value).map(it => it.props().children);

        expect(allHeaderValues).toEqual(['MINIFIED', 'MINIFIED + GZIPPED', 'UNPACKED']);
        expect(allStatsValues).toEqual(["-", "1.45 MB", "-",]);
    });

    it("should pass the snapshot test", () => {
        runTheSnapshotTest(<SizeStatisticsPanel {...defaultTestProps} />)
    });

});