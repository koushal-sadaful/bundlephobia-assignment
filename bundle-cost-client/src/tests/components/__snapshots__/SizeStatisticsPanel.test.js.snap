// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SizeStatisticsPanel should pass the snapshot test 1`] = `
<div
  className="statistics-panel"
>
  <div
    className="ui cards"
  >
    <div
      className="ui card"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="ui statistic"
        >
          <div
            className="value"
          >
            2.12 MB
          </div>
        </div>
        <div
          className="header"
        >
          MINIFIED
        </div>
      </div>
    </div>
    <div
      className="ui card"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="ui statistic"
        >
          <div
            className="value"
          >
            1.45 MB
          </div>
        </div>
        <div
          className="header"
        >
          MINIFIED + GZIPPED
        </div>
      </div>
    </div>
    <div
      className="ui card"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="ui statistic"
        >
          <div
            className="value"
          >
            41.5 MB
          </div>
        </div>
        <div
          className="header"
        >
          UNPACKED
        </div>
      </div>
    </div>
  </div>
</div>
`;
